src/metalperformanceshadersgraph.cs

```diff
@@ -0,0 +1,1840 @@
+using System;
+using CoreGraphics;
+using Foundation;
+using Metal;
+using MetalPerformanceShaders;
+using ObjCRuntime;
+
+#if !NET
+using NativeHandle = System.IntPtr;
+#endif
+
+using MPSGraphTensorDataDictionary = Foundation.NSDictionary<MetalPerformanceShadersGraph.MPSGraphTensor, MetalPerformanceShadersGraph.MPSGraphTensorData>;
+using MPSGraphTensorShapedTypeDictionary = Foundation.NSDictionary<MetalPerformanceShadersGraph.MPSGraphTensor, MetalPerformanceShadersGraph.MPSGraphShapedType>;
+
+
+// MPSGraphType was introduced in iOS 15 (macOS 12) and became the base class for
+// MPSGraphShapedType which exists in iOS 14.
+// In iOS 14, MPSGraphShapedType inherits directly from NSObject.
+// To keep compatibility, MPSGraphType is type-erased to just NSObject
+// and implementers have to add NSCopying.
+// @interface MPSGraphType : NSObject
+// [TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+// [BaseType (typeof(NSObject))]
+// interface MPSGraphType : NSCopying
+// {
+// }
+using MPSGraphType = Foundation.NSObject;
+
+
+namespace MetalPerformanceShadersGraph
+{
+	// MPSGraph.h
+
+	[iOS (14,0), TV (14,0), Mac (11,0), MacCatalyst (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface MPSGraph {
```

Comment:
Do you know if the default ctor in Objective-C is allowed? the alloc/init dance

Reply:
Yes. And you need to be able to inherit from this class (as per their examples). I've confirmed it working on iOS and Mac. They're good about marking classes without a default init. The `new` static is redundant...